sig-scalability/slos/dns_latency.md

```diff
@@ -0,0 +1,44 @@
+## In-cluster dns latency SLIs/SLOs details
+
+### Definition
+
+| Status | SLI | SLO |
+| --- | --- | --- |
+| __WIP__ | In-cluster dns latency from a single prober pod, measured as latency of per second DNS lookup for "null service" from that pod, measured as 99th percentile over last 5 minutes. | In default Kubernetes installataion with RTT between nodes <= Y, 99th percentile of (99th percentile over all prober pods) per cluster-day <= X |
+
+### User stories
+- As a user of vanilla Kubernetes, I want some guarantee how fast my in-cluster
+DNS requests are resolved
+
+### Other notes
+- We obviously can't give any guarantee in a general case, because cluster
+administrators may configure cluster as they want.
+- As a result, we define the SLI to be very generic (no matter how your cluster
+is set up), but we provide SLO only for default installations with an additional
+requirement that low-level RTT between nodes is lower than Y.
+- DNS latency is one of the most crucial aspects from the point of view
```

Comment:
I'd argue DNS Latency is really part of a pair of critical metrics, the other being "drop rate" or "timeout rate".  Obviously this is much harder to sample, since it is already a very low (but not zero) number.  Different discussion?

Reply:
I would prefer move it to a different discussion (I will add a TODO in this doc tomorrow).
Is that fine for you?

Reply:
average and p99? maybe p99.9, if using 5m sliding windows?

Reply:
Moving to a different conversation is fine.  I think we'd need to probe a lot more often to detect it reliably.  At least, I hope so!!

Reply:
p99 and p99 and 5m is consistent with all other SLIs we have (so I would like to change to average and 5m).
Also, given that it's probing over 5m every 1s, p99,9 means max, right (there are less than 1k probes in 5m).

Reply:
TODO added below.